api/app/Services/Api/Vanilla/Client.php

```diff
@@ -0,0 +1,283 @@
+<?php
+
+namespace App\Services\Api\Vanilla;
+
+use InvalidArgumentException;
+use Guzzle\Http\Client as GuzzleClient;
+use Guzzle\Http\Exception\RequestException;
+use Symfony\Component\HttpKernel\Exception\ServiceUnavailableHttpException;
+
+class Client
+{
+    /**
+     * API base URL.
+     *
+     * @var string
+     */
+    protected $baseUrl;
+
+    /**
+     * API secret key.
+     *
+     * @var string
+     */
+    protected $secret;
+
+    /**
+     * Default request headers.
+     *
+     * @var array
+     */
+    protected $headers = ['content-type' => 'application/json'];
+
+    /**
+     * The Guzzle instance used to communicate with the API.
+     *
+     * @var GuzzleClient
+     */
+    protected $client;
+
+    /**
+     * The user to operate with. Defaults to Vanilla's internal system user.
+     *
+     * @var array
+     */
+    protected $user = [
+        'username' => 'system',
+        'email' => 'system@domain.com',
+    ];
+
+    /**
+     * Map group name to class names.
+     *
+     * @var array
+     */
+    protected $map = [
+        'configuration' => 'Configuration',
+        'comments' => 'Comment',
+        'discussions' => 'Discussion',
+        'users' => 'User',
+    ];
+
+    /**
+     * Assign dependencies.
+     *
+     * @param  GuzzleClient $client
+     *
+     * @return void
+     */
+    public function __construct(GuzzleClient $client)
+    {
+        $this->client = $client;
+
+        $this->secret = getenv('VANILLA_API_SECRET');
+
+        // If we run in a continous integrated enviroment for testing, the forum
+        // server will be running on a CLI server which doesn't support pretty
+        // urls, so in that case we use another base url.
+        $base = getenv('CI') ? '/index.php' : '/api/';
+        $this->baseUrl = getenv('FORUM_HOST').$base;
```

Comment:
Nice approach, for your reference you can see how Lumen does it here https://github.com/spira/spira/blob/master/api/server.php